std/range/package.d

```diff
@@ -5665,16 +5768,6 @@ pure nothrow @nogc @safe unittest
     assert(iota(ssize_t.max, ssize_t.min, -3).length == size_t.max / 3);
 }
 
-debug @system unittest
-{//check the contracts
-    import core.exception : AssertError;
-    import std.exception : assertThrown;
-    assertThrown!AssertError(iota(1,2,0));
```

Comment:
To unify the interface, I opted to remove the assertions that are tested here. The other overloads did not assert, and removing assertions break less code than adding them.